libbeat/common/kubernetes/metadata.go

```diff
@@ -102,9 +110,27 @@ func (g *metaGenerator) ResourceMetadata(obj Resource) common.MapStr {
 	}
 
 	annotationsMap := generateMapSubset(accessor.GetAnnotations(), g.IncludeAnnotations, g.AnnotationsDedot)
-	meta := common.MapStr{}
+	metadata := common.MapStr{}
 	if accessor.GetNamespace() != "" {
-		meta["namespace"] = accessor.GetNamespace()
+		namespase := accessor.GetNamespace()
+
+		ns, err := g.k8sClient.CoreV1().Namespaces().Get(namespase, metav1.GetOptions{})
```

Comment:
This will be called every time we want to enrich a log line or metrics event. This line would result on way many calls to the API, isn't it? We probably need to cache these to make sure we don't kill API server.

What do you think about using a shared informer to keep a list of Namespaces? That should solve this